packages/framework/client-logger/fluid-telemetry/src/test/containerTelemetryEndToEnd.spec.ts

```diff
@@ -0,0 +1,253 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TinyliciousClient } from "@fluidframework/tinylicious-client/internal";
+import { SharedTree } from "@fluid-experimental/tree";
+import * as AppInsights from "@microsoft/applicationinsights-web";
+import type Sinon from "sinon";
+import { spy } from "sinon";
+import { expect } from "chai";
+import { startTelemetry } from "../factory/index.js";
+import {
+	AppInsightsTelemetryConsumer,
+	ContainerTelemetryEventNames,
+	type ContainerConnectedTelemetry,
+	type ContainerDisconnectedTelemetry,
+	type ContainerDisposedTelemetry,
+	type IContainerTelemetry,
+} from "../index.js";
+
+// This test suite creates an actual IFluidContainer and confirms events are fired with the expected names during the expected events
+
+describe("container telemetry E2E", () => {
+	let schema;
+	let tinyliciousClient: TinyliciousClient;
+	let appInsightsClient: AppInsights.ApplicationInsights;
+	let appInsightsTrackEventSpy: Sinon.SinonSpy;
+
+	before(() => {
+		tinyliciousClient = new TinyliciousClient({ connection: { port: 7070 } });
+		schema = {
+			initialObjects: {
+				sharedMap1: SharedTree,
+			},
+		};
+
+		appInsightsClient = new AppInsights.ApplicationInsights({
+			config: {
+				connectionString:
+					// (this is an example string)
+					"InstrumentationKey=abcdefgh-ijkl-mnop-qrst-uvwxyz6ffd9c;IngestionEndpoint=https://westus2-2.in.applicationinsights.azure.com/;LiveEndpoint=https://westus2.livediagnostics.monitor.azure.com/",
+			},
+		});
+		appInsightsTrackEventSpy = spy(appInsightsClient, "trackEvent");
+	});
+
+	beforeEach(() => {
+		// We need to reset the Application Insights client spy for each test so the trackEvent calls
+		// from the last test don't bleed into the next one.
+		appInsightsTrackEventSpy.resetHistory();
+	});
+
+	it("IFluid container's 'connected' system event produces expected ContainerConnectedTelemetry using AppInsightsTelemetryConsumer", async () => {
+		const { container } = await tinyliciousClient.createContainer(schema);
+
+		const containerId = await container.attach();
+		startTelemetry({
+			container,
+			containerId,
+			consumers: [new AppInsightsTelemetryConsumer(appInsightsClient)],
```

Comment:
I think we should simplify a bit how much this test wants to cover. We could plug a fake consumer, a small class implemented in this test file, that logs what it receives to an array or something we can inspect, so we don't tie these tests to implementation details of the AppInsightsClient (e.g. the fact that it uses a `trackEvent` method; if they later change that, these tests shouldn't need any updates). That still lets us verify that the event looks like we expect it to.

I think that's probably also a good idea for the other tests file. And if we want we could have unit tests on the AppInsightsTelemetryConsumer itself, that check that when passed some particular event, it makes a call to `trackEvent` with an argument that looks exactly as expected (i.e. the eventName gets copied to a "name" property, and everything else is identical).

Reply:
Updated to use a simple telemetry consumer class purely for testing